--- a/drivers/usb/serial/usb_wwan.c
+++ b/drivers/usb/serial/usb_wwan.c
@@ -478,6 +478,22 @@
 			  usb_sndbulkpipe(serial->dev, endpoint) | dir,
 			  buf, len, callback, ctx);
 
+#if 1 //Added by Quectel for zero packet
+	if (dir == USB_DIR_OUT) {
+	struct usb_device_descriptor *desc = &serial->dev->descriptor;
+		if (desc->idVendor == cpu_to_le16(0x2C7C))
+		urb->transfer_flags |= URB_ZERO_PACKET;
+		}
+#endif
+
+	/*start add by fibocom*/
+    if (dir == USB_DIR_OUT) {
+    struct usb_device_descriptor *desc = &serial->dev->descriptor;
+        if (desc->idVendor == cpu_to_le16(0x2cb7))
+        urb->transfer_flags |= URB_ZERO_PACKET;
+        }
+    /*end add by fibocom*/
+
 	if (intfdata->use_zlp && dir == USB_DIR_OUT)
 		urb->transfer_flags |= URB_ZERO_PACKET;
 
