--- a/drivers/net/usb/qmi_wwan.c	
+++ b/drivers/net/usb/qmi_wwan.c
@@ -1047,8 +1047,11 @@
 	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0620)},	/* Quectel EM160R-GL */
 	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0800)},	/* Quectel RM500Q-GL */
 	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0801)},	/* Quectel RM520N */
+	{QMI_MATCH_FF_FF_FF(0x2dee, 0x4D22)},	/* MeigLink SRM815  */
+	{QMI_MATCH_FF_FF_FF(0x05c6, 0x90D5)},	/* ChinaMobile F03X SDX55 MBIM*/
 
 	/* 3. Combined interface devices matching on interface number */
+	{QMI_QUIRK_SET_DTR(0x05c6, 0x9091, 2)},	/* ChinaMobile F03X SDX55 QMI */
 	{QMI_FIXED_INTF(0x0408, 0xea42, 4)},	/* Yota / Megafon M100-1 */
 	{QMI_FIXED_INTF(0x05c6, 0x6001, 3)},	/* 4G LTE usb-modem U901 */
 	{QMI_FIXED_INTF(0x05c6, 0x7000, 0)},
